tools/javadoc/javadoc.bzl

```diff
@@ -24,7 +24,12 @@ def _android_jar(android_api_level):
 def _javadoc_library(ctx):
     _check_non_empty(ctx.attr.root_packages, "root_packages")
 
-    transitive_deps = [dep.java.transitive_deps for dep in ctx.attr.deps]
+    transitive_deps = [
+        dep[JavaInfo].transitive_deps if JavaInfo in dep else dep.java.transitive_deps
```

Comment:
This list comprehension is now a little too complex. Can you convert it into a regular loop?

Reply:
Sure, np.